language-support/scala/codegen-sample-app/src/test/scala/com/digitalasset/codegen/GeneratedCommandsUT.scala

```diff
@@ -30,4 +30,17 @@ class GeneratedCommandsUT extends WordSpec with Matchers with Inside {
       }
     }
   }
+
+  "key" should {
+    "make an exercise-by-key command" in {
+      inside((KeyedNumber key alice exerciseIncrement (alice, 42)).command.command) {
+        case rpccmd.Command.Command.ExerciseByKey(
+            rpccmd.ExerciseByKeyCommand(Some(tid), Some(k), "Increment", Some(choiceArg))) =>
+          import com.daml.ledger.client.binding.Value.encode
+          tid should ===(KeyedNumber.id)
+          k should ===(encode(alice))
+          choiceArg should ===(encode(Increment(42)))
+      }
+    }
+  }
```

Comment:
Would it make sense to verify the behavior of a failing exercise-by-key in a test?

Reply:
No, because the result of the command has nothing to do with the codegen.

Reply:
I think testing for a properly generated `Command.Command.ExerciseByKey` is sufficient (love the name Command.Command :)) It compiles and all required field are properly populated. Making sure that the command is accepted or rejected by the ledger is NOT the concern of the scala codegen.